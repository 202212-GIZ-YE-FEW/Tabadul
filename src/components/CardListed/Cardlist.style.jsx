import styled from "@emotion/styled";
import Image from "next/image";

export const Container = styled.div`
    margin-left: auto;
    margin-right: auto;
    padding-left: 15px;
    padding-right: 15px;
    @media (min-width: 768px) {
        width: 750px;
    }
    /* Medium */
    @media (min-width: 992px) {
        width: 970px;
    }
    @media (min-width: 1200px) {
        width: 1370px;
    }
    /* Large */
    @media (min-width: 1500px) {
        width: 1400px;
    }
`;

export const BoxCategore = styled.div`
    display: flex;
    justify-content: space-around;
    align-items: center;
`;
export const BoxCard = styled.div`
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    gap: 85px 18px;
    margin-top: 100px;
`;
export const Cards = styled.div`
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    gap: 20px;
`;

export const Card = styled.div`
    // position: absolute;
    background: #ffffff;
    box-shadow: 0px 0px 7px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    background: #ffffff;
`;

export const Cardimg = styled(Image)`
    border-radius: 19px 19px 0px 0px;
    max-width: 100%;
`;

export const Cardinfo = styled.div`
    display: flex;
    align-items: center;
    margin-left: 10px;
`;

export const LocationTitle = styled.p`
    // font-family: "Poppins";
    font-weight: 400;
    font-size: 20px;
    line-height: 30px;
    text-align: center;
    margin-left: 10px;
`;

export const CardHeader = styled.p`
    // font-family: "Poppins";
    font-weight: 700;
    font-size: 20px;
    line-height: 30px;
    color: #3c4347;
    margin-left: 10px;
    margin-top: 0;
`;
export const CardParagraph = styled.p`
    // font-family: "Poppins";
    font-size: 16px;
    font-weight: 400;
    width: 200px;
    color: #3c4347;
    margin-left: 10px;
    margin-top: 0;
`;
export const InputContainer = styled.form`
    position: relative;
    width: 516px;
    height: 54px;
    margin: auto auto 50px auto;
    @media screen and (max-width: 786px) {
        width: 350px;
        max-width: 100%;
        height: 42px;
    }
`;

export const SerachBar = styled.input`
    width: 100%;
    height: 100%;
    display: block;
    box-shadow: 0px 5px 6px rgba(0, 0, 0, 0.25);
    border-radius: 69px;
    border: none;
    margin-bottom: 40px;
    padding-left: 39px;
    font-size: 24px;
    line-height: 36px;
    &:focus {
        outline: none;
    }
    &::placeholder {
        color: #bebebe;
    }
    @media screen and (max-width: 768px) {
        background: #ffffff;
        box-shadow: 0px 5px 6px rgba(0, 0, 0, 0.25);
        border-radius: 69px;
        padding-left: 21.67px;
        font-size: 20px;
        line-height: 30px;
    }
`;
export const SerachBarImage = styled(Image)`
    position: absolute;
    top: 50%;
    right: 15.38px;
    transform: translateY(-50%);
    width: 32.21px;
    height: 32.84px;
    z-index: 99;
    @media screen and (max-width: 768px) {
        width: 32px;
    }
`;
