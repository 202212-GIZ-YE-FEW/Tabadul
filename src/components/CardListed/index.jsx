import React, { useState } from "react";

import {
    BoxCard,
    Card,
    CardHeader,
    Cardimg,
    Cardinfo,
    CardParagraph,
    Container,
    LocationTitle,
} from "./Cardlist.style";
import Categories from "../Categories";
import { CARDS } from "../../../data/db";
function CardListed() {
    const [selectedCategory, setSelectedCategory] = useState("All");

    const itemDisplay = CARDS.filter((item) => {
        if (selectedCategory === "All") return true;

        return selectedCategory === item.category;
    });
    return (
        <Container>
            <Categories setSelectedCategory={setSelectedCategory} />
            <BoxCard>
                {itemDisplay.map((ele) => {
                    return (
                        <Card key={ele.id}>
                            <Cardimg
                                src='/images/Cardimg.svg'
                                alt='CardImage'
                                width={258.4}
                                height={185.44}
                            />
                            <Cardinfo>
                                <LocationTitle>Istanbol</LocationTitle>
                            </Cardinfo>
                            <CardHeader>{ele.category}</CardHeader>
                            <CardParagraph>
                                Short description for the available item{" "}
                            </CardParagraph>
                        </Card>
                    );
                })}
            </BoxCard>
        </Container>
    );
}

export default CardListed;
