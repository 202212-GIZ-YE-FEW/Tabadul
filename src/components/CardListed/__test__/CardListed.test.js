import renderer from "react-test-renderer";

import CardListed from "..";

it("renders correctly", () => {
    const tree = renderer.create(<CardListed />).toJSON();
    expect(tree).toMatchSnapshot();
});
