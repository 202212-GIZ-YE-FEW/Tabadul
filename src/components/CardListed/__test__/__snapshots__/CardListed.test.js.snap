// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-pp2h56"
>
  <div
    className="css-1y9f80n"
  >
    <div
      className=""
    >
      <div
        className="shimmer shimmer-thumbnail border-rounded css-1quxnu4"
        style={
          Object {
            "height": "250px",
            "width": "240px",
          }
        }
      />
    </div>
    <div
      className=""
    >
      <div
        className="shimmer shimmer-thumbnail border-rounded css-1quxnu4"
        style={
          Object {
            "height": "250px",
            "width": "240px",
          }
        }
      />
    </div>
    <div
      className=""
    >
      <div
        className="shimmer shimmer-thumbnail border-rounded css-1quxnu4"
        style={
          Object {
            "height": "250px",
            "width": "240px",
          }
        }
      />
    </div>
    <div
      className=""
    >
      <div
        className="shimmer shimmer-thumbnail border-rounded css-1quxnu4"
        style={
          Object {
            "height": "250px",
            "width": "240px",
          }
        }
      />
    </div>
    <div
      className=""
    >
      <div
        className="shimmer shimmer-thumbnail border-rounded css-1quxnu4"
        style={
          Object {
            "height": "250px",
            "width": "240px",
          }
        }
      />
    </div>
  </div>
</div>
`;
